content/apollo.md

```diff
@@ -0,0 +1,102 @@
+
+Mutators are only run on the server, but they can return an object which then can update the local cache without the need to call a query again.
+
+<h3 id="server">Apollo Server</h3>
+
+[Apollo server docs](https://www.apollographql.com/docs/apollo-server/)
+
+<h4 id="getting-data-server">Getting data</h4>
+
+Instead of using `Meteor.publish` to define publications, you write [resolve functions](https://www.apollographql.com/docs/apollo-server/data/resolvers/) – called *resolvers* – that fetch different types of data in the query.
+
+<h4 id="changing-data-server">Changing data</h4>
+
+Instead of using `Meteor.methods` to define methods, you write [mutators](https://www.apollographql.com/docs/tutorial/mutation-resolvers/) – functions that *mutate* (change) data.
+
+These are part of the resolver functions under `Mutation` key.
+
+<h3 id="graphql">GraphQL</h3>
+
+GraphQL is a query language for apps to get the data they want. Instead of the server deciding what's in a publication, the client uses GraphQL to say exactly which fields of which objects it wants. 
+
+- [About GraphQL](https://graphql.org/)
+- [Intro to GraphQL](https://medium.com/apollo-stack/the-basics-of-graphql-in-5-links-9e1dc4cac055)
+- [GraphQL coming from REST](https://medium.com/apollo-stack/how-do-i-graphql-2fcabfc94a01#.pfdj5bxxj)
+
+<h3 id="advanced">Advanced<h3>
+
+[Principled GraphQL](https://principledgraphql.com/)
+
+<h4 id="latency">Latency</h4>
+
+Meteor publications are blocking, whereas multiple GraphQL queries are executed in parallel. Publications stream data to the client as it arrives, whereas all the resolvers in a GraphQL query have to return before the data is sent to the client. (Although GraphQL is discussing adding the ability to stream results to the client as they come in.)
+
+<h3>Meteor specific</h3>
+
+Meteor has a specific Apollo package which includes user object into the context of a query.
+
+```shell
+meteor add apollo
+```
+
+On server you import `getUser` function and include it into the context option when setting up Apollo server:
+
+```javascript
+import { ApolloServer } from 'apollo-server-express';
+import { getUser } from 'meteor/apollo';
+import typeDefs from '/imports/apollo/schema.graphql';
+import { resolvers } from '/server/resolvers';
+
+const server = new ApolloServer({
+  typeDefs,
+  resolvers,
+  context: async ({ req }) => ({
+    user: await getUser(req.headers.authorization)
+  })
+});
+```
+
+This will make user data available (if user is logged in) as the option in the query:
+```javascript
+{
+  Query: {
+    userUniverses: async (obj, { hideOrgs }, { user }) => {
+      if (!user) return null
+      const selector = { userId: user._id, }
+      if (hideOrgs) selector.organizationId = { $exists: false }
+      return UniversesCollection.find(selector).fetch()
+    }
+  }
+}
+```
```

Comment:
It would be helpful to list other packages, like https://github.com/quavedev/graphql that provides more features.

Reply:
Added, but it would be nice if it was updated for the latest version of Apollo.